std/functional.d

```diff
@@ -1814,13 +1814,42 @@ if (isCallable!(F))
     {
         return fp;
     }
+    else static if (is(F Func == Func*) && is(Func == function) && is(Func Params == __parameters))
+    {
+        return () @trusted
+        {
+            return buildDelegate(fp);
+        }();
+    }
     else static if (is(typeof(&F.opCall) == delegate)
                 || (is(typeof(&F.opCall) V : V*) && is(V == function)))
     {
         return toDelegate(&fp.opCall);
     }
     else
     {
+        return buildDelegate(fp);
```

Comment:
... what type of objects hit this branch and why is that not a safe (i.e. trustable) thing to do?

Reply:
I’ve replaced this branch with a `static assert(false);` – as I couldn’t think of a case that would both match the parameter and satisfy the template constraint. What do you think about that?